app/lib/provider_discovery.rb

```diff
@@ -22,6 +22,7 @@ def discover_provider(url, options = {})
         format ||= :xml if provider_endpoint
       end
 
+      raise OEmbed::NotFound, url unless provider_endpoint
```

Comment:
There are many `if val.nil?` In this repository. I think that the same way is better